### DISCUSSION TITLE
The problem of using of d3.hierarchy

### DISCUSSION THREAD

**Discussion by catbao** [d3/d3#3561] [Q&A] [answered]
If I want to use the pixels of the scatter plot as the basic unit, and construct the kd-tree according to the number of scatter points in different pixel areas, is it ok to use d3.hierarchy?
It may be like this:
![4U%8)})9O64 @F(X36 )8{W](https://user-images.githubusercontent.com/52957986/159152981-384f6f05-42f2-4c90-8d55-cbfe029c6a39.png)
Very thanks to answer my question !!!!!! Very thanks!

**Comment by git-ashish** [comment, marked as answer]
Whether to use d3.hierarchy or not would be based your choice which depends on the fact if the generated outputs are acceptable to your use case. Hierarchy would simply create an hierarchy based on quadrants and the structure as measured in b) and c). If you can establish a hierarchy relationship to measure, you can certainly use d3.hierarchy. Hope it helps

**Reply by catbao** [reply]
Thank you for your answer!  I also have a question that in d3's API,  how to get the number of points in every pixel or pixel areas just like in pic(a) to pic(b)? Or is this possible in d3?

**Reply by git-ashish** [reply]
I don't know what is being done in c) but you could divide your space in rows and columns and assign every point to a cell. Having information per cell, you could derive e)

**Reply by catbao** [reply]
Thanks for your answer again! Because I'm not so familier with d3, so my question my be navie. I understand what you mean, my space is a pixel space, the process from a to b is to calculate the number of scatter points in each pixel and get the number on each pixel in b, I want to know if d3 can do this calculation, just to get the count numbers in every pixel?

**Reply by git-ashish** [reply]
You will need to do those calculations yourself. Close to what you desire would be [d3.bin](https://observablehq.com/@d3/d3-bin)

**Reply by catbao** [reply]
Ok I know，very thanks for you!
